pkg/agent/flowexporter/exporter.go

```diff
@@ -353,11 +395,57 @@ func (exp *FlowExporter) fillEgressInfo(conn *connection.Connection) {
 	}
 }
 
+// getServiceName returns the name of the service from the set of services provided
+// which has the matching provided port and the name of the port.
+// Empty strings are returned if no service match is found.
+func getServiceName(port uint16, services []*corev1.Service) (string, string) {
+	for _, service := range services {
+		for _, servicePort := range service.Spec.Ports {
+			if servicePort.Port == int32(port) {
+				return service.Name, servicePort.Name
+			}
+		}
+	}
+
+	return "", ""
+}
+
+// FillServiceInfo updates the given conn of type FlowTypeToExternal with the name of
+// the service whos port matches the destination port. An error is returned if no match
+// is found or errors occurred retrieving services
+func (exp *FlowExporter) FillServiceInfo(conn *connection.Connection) error {
```

Comment:
I don't understand this function. It tries to map a port to a Service? You can have 100s of Service sharing the same port value, as only `ClusterIP:port` can uniquely identify a Service, not `port` by itself.
Now, if we are considering `NodePort`, that's different, as it will uniquely identify the Service, but that doesn't seem to be what is happening here.

On a side note, this is a pretty inefficient implementation. We usually rely on an https://pkg.go.dev/k8s.io/client-go/tools/cache#Indexer when we want to map a property value to a list of matching objects (many examples in the code base).